druid/src/core.rs

```diff
@@ -957,6 +977,25 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
                     f.call(&self.state);
                     true
                 }
+                InternalLifeCycle::RouteDisabledChanged => {
+                    let was_disabled = self.state.is_disabled();
+
+                    self.state.is_explicitly_disabled = self.state.is_explicitly_disabled_new;
+
+                    if was_disabled != self.state.is_disabled() {
+                        // In case we change but none of our children we still need to update the
+                        // focus-chain
+                        self.state.reset_focus_chain();
+                        extra_event = Some(LifeCycle::DisabledChanged(self.state.is_disabled()));
+                        //Each widget needs only one of DisabledChanged and RouteDisabledChanged
+                        false
+                    } else if self.state.children_disabled_changed {
+                        self.state.reset_focus_chain();
+                        true
+                    } else {
+                        false
+                    }
```

Comment:
okay trying to wrap my head around this.

Can we simplify this?

- *always* clear the focus chain
- *always* fixup the focus chain afterwards?

I'm also curious about the logic here. I would expect it to be true that in order to see this event, *either* we have been explicitly disabled, *or* some of our children are, and the final branch here should be unreachable?

Reply:
Basically: looking back through this method, I am struck by how simple it is to handle the focus_chain stuff currently, and I feel like it should be possible to have the new logic be similarly simple? This all comes down to understanding our invariants; what guarantees can we make about our state before we recurse, and what guarantees can we expect afterwards?

Reply:
If we always clear the focus_chain we need to route this event to all widgets in the window. I could change it if you think that's better. 
In my implementation i always clear the focus-chain if i recurse the event or send the extra event.